Userland/js.cpp

```diff
@@ -36,11 +37,79 @@
 #include <LibJS/Runtime/Value.h>
 #include <stdio.h>
 
-#define PROGRAM 6
+bool dump_ast = false;
+
+String read_next_piece()
+{
+    StringBuilder piece;
+    int level = 0;
+
+    do {
+        if (level == 0)
+            fprintf(stderr, "> ");
+        else
+            fprintf(stderr, ".%*c", 4 * level + 1, ' ');
+
+        char* line = nullptr;
+        size_t allocated_size = 0;
+        ssize_t nread = getline(&line, &allocated_size, stdin);
+        if (nread < 0) {
+            if (errno == ESUCCESS) {
+                // Explicit EOF; stop reading. Print a newline though, to make
+                // the next prompt (or the shell prompt) appear on the next
+                // line.
+                printf(stderr, "\n");
```

Comment:
This should be `fprintf` :)

Reply:
Oops, that's what I get for not testing "obviously correct" fixes :)

Reply:
Np I'll just merge it manually with a fix :)